[k8s] snippet on multi node jobs in k8s (#5495)
skypilot-org/skypilot: docs/source/reference/kubernetes/kubernetes-getting-started.rst

        curl -LO "https://dl.k8s.io/release/v1.28.11/bin/linux/amd64/kubectl" && \
        sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl && \
        echo 'export PATH="$PATH:$HOME/.local/bin"' >> ~/.bashrc

Multi-node jobs

> ----------------

Author: Thanks for the suggestion, updated as shown
<img width="828" alt="Screenshot 2025-05-06 at 10 17 57 AM" src="https://github.com/user-attachments/assets/2d920ffc-62cd-4d43-a634-fddce4d4f9e2" />

> :ref:`Multi-node jobs <dist-jobs>` are supported on Kubernetes. When a multi-node job is launched, each node in a SkyPilot cluster is provisioned as a separate pod.

Reviewer: ```suggestion
:ref:`Multi-node jobs <dist-jobs>` are supported on Kubernetes. When a multi-node job is launched, each "node" in the SkyPilot cluster is provisioned as a separate pod.
```
Helps clarify that we are not talking about actual "nodes" here as in the Kubernetes sense.

> SkyPilot will attempt to place each pod on a different node in the cluster.

Reviewer: ```suggestion
SkyPilot will attempt to place each pod on a different underlying node in the Kubernetes cluster.
```

> SkyPilot will try to schedule all pods on a given cluster. If SkyPilot cannot schedule all pods on a given cluster (i.e. some or all of the pods cannot be scheduled),
> SkyPilot will fail over to another cluster or cloud.

Reviewer: ```suggestion
If you have multiple Kubernetes clusters, SkyPilot will schedule all the job pods in the same cluster. If SkyPilot cannot schedule all pods on a given Kubernetes cluster (i.e. some or all of the pods cannot be scheduled due to insufficient resources),
it will fail over to another Kubernetes cluster or cloud.
```
